### ISSUE TITLE
DOC: Add the internal data model of metadata

### ISSUE THREAD

**Issue by adswa** [datalad/datalad-metalad#315]
@christian-monch shared an internal google doc that contained the architecture brainstorming and spec of MetaLad's metadata model. I believe that an overview of this architecture with prettified SVG-based graphics would enhance the docs a lot, and would probably be very valuable to include in grant reports or audits for software security clearances.

**Comment by christian-monch** [comment]
Good idea and very good point about the grant reports and audits! Thx @adswa
